swap/simulations_test.go

```diff
@@ -315,13 +318,17 @@ func TestMultiChequeSimulation(t *testing.T) {
 		// the node has all other peers in its peer list
 		debitorSvc.swap.peersLock.Lock()
 		debSwapLen = len(debitorSvc.swap.peers)
-		debLen = len(debitorSvc.peers)
 		debitorSvc.swap.peersLock.Unlock()
+		debitorSvc.lock.Lock()
+		debLen = len(debitorSvc.peers)
+		debitorSvc.lock.Unlock()
 
 		creditorSvc.swap.peersLock.Lock()
 		credSwapLen = len(creditorSvc.swap.peers)
-		credLen = len(creditorSvc.peers)
 		creditorSvc.swap.peersLock.Unlock()
+		creditorSvc.lock.Lock()
+		credLen = len(creditorSvc.peers)
+		creditorSvc.lock.Unlock()
```

Comment:
If it can't be done otherwise, it must be done like this, of course. But I must say, the code becomes very ugly here... And I can imagine that any dev who has to make something similar like this will make mistakes here and/or will not understand directly what is done here

Reply:
I agree, but I don't think it can be avoided here, at least not without significant change to the test itself.